src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -0,0 +1,273 @@
+        jit_operator->add_jit_operator(std::make_shared<JitCompute>(expression));
+      }
+      write_table->add_output_column(output_column.get<0>(), expression->result());
+    }
+    jit_operator->add_jit_operator(write_table);
+    return jit_operator;
+  } catch (const std::exception& exception) {
+    // If JitOperatorWrapper creation fails, we fall back to the regular LQPTranslator
+    return LQPTranslator::translate_node(node);
+  }
+}
+
+std::shared_ptr<const JitExpression> JitAwareLQPTranslator::_translate_to_jit_expression(
+    const std::shared_ptr<AbstractLQPNode>& node, JitReadTuple& jit_source,
+    const std::shared_ptr<AbstractLQPNode>& input_node) const {
+  std::shared_ptr<const JitExpression> left, right;
+  switch (node->type()) {
+    case LQPNodeType::Predicate:
+      // If there are further conditions (either PredicateNodes of UnionNodes) down the line, we need a logical AND here
+      if (_has_another_condition(node->left_input())) {
+        left = _translate_to_jit_expression(std::dynamic_pointer_cast<PredicateNode>(node), jit_source, input_node);
+        right = _translate_to_jit_expression(node->left_input(), jit_source, input_node);
+        return std::make_shared<JitExpression>(left, ExpressionType::And, right, jit_source.add_temporary_value());
+      } else {
+        return _translate_to_jit_expression(std::dynamic_pointer_cast<PredicateNode>(node), jit_source, input_node);
+      }
+
+    case LQPNodeType::Union:
+      left = _translate_to_jit_expression(node->left_input(), jit_source, input_node);
+      right = _translate_to_jit_expression(node->right_input(), jit_source, input_node);
+      return std::make_shared<JitExpression>(left, ExpressionType::Or, right, jit_source.add_temporary_value());
+
+    case LQPNodeType::Projection:
+      // We don't care about projection nodes here, since they do not perform any tuple filtering
+      return _translate_to_jit_expression(node->left_input(), jit_source, input_node);
+
+    default:
+      Fail("Unreachable.");
+  }
+}
+
+std::shared_ptr<const JitExpression> JitAwareLQPTranslator::_translate_to_jit_expression(
+    const std::shared_ptr<PredicateNode>& node, JitReadTuple& jit_source,
+    const std::shared_ptr<AbstractLQPNode>& input_node) const {
+  auto condition = predicate_condition_to_expression_type.at(node->predicate_condition());
+  auto left = _translate_to_jit_expression(node->column_reference(), jit_source, input_node);
+  std::shared_ptr<const JitExpression> right;
+
+  switch (condition) {
+    /* Binary predicates */
+    case ExpressionType::Equals:
+    case ExpressionType::NotEquals:
+    case ExpressionType::LessThan:
+    case ExpressionType::LessThanEquals:
+    case ExpressionType::GreaterThan:
+    case ExpressionType::GreaterThanEquals:
+    case ExpressionType::Like:
+    case ExpressionType::NotLike:
+      right = _translate_to_jit_expression(node->value(), jit_source, input_node);
```

Comment:
Is `right` the right hand side of the comparison? Usually, when we talk about LQPs, left and right are tables. Maybe that's where my confusion comes from.

Reply:
Yes. What alternative would you prefer? `left_side` / `left_input` / `left_child` / `lhs`?